### PR TITLE
Split contents of dom.js among four(4) files

### LINKED ISSUE #7308: [p5.js 2.0 RFC Proposal]: dom.js file separation
### Increasing access

This proposal is to split the large [dom.js](https://github.com/processing/p5.js/blob/main/src/dom/dom.js) file which is nearly 6,000 lines, into smaller files to make the functions within more manageable and usable.

### Which types of changes would be made?

- [ ] Breaking change (Add-on libraries or sketches will work differently even if their code stays the same.)
- [ ] Systemic change (Many features or contributor workflows will be affected.)
- [X] Overdue change (Modifications will be made that have been desirable for a long time.)
- [ ] Unsure (The community can help to determine the type of change.)

### Most appropriate sub-area of p5.js?

- [ ] Accessibility
- [ ] Color
- [ ] Core/Environment/Rendering
- [ ] Data
- [X] DOM
- [ ] Events
- [ ] Image
- [ ] IO
- [ ] Math
- [ ] Typography
- [ ] Utilities
- [ ] WebGL
- [ ] Build process
- [ ] Unit testing
- [ ] Internationalization
- [ ] Friendly errors
- [ ] Other (specify if possible)

### What's the problem?

[dom.js](https://github.com/processing/p5.js/blob/main/src/dom/dom.js) is nearly 6,000 lines of code and splitting it up can make the code within more manageable. 

### What's the solution?

This file is already organized into several categories. This proposal would see it split into 4+ files including the current dom.js file depending on how the community feels. The split would look as follows:

* dom.js:
   * Lines 21-107:   P5.prototype.select
   * Lines 109-204: p5.prototype.selectAll
   * Lines 206-219: p5.prototype._getContainer
   * Lines 221-254: p5.prototype._wrapElement
   * Lines 256-328: p5.prototype._removeElements
   * Lines 478-489: p5.prototype.addElement 
   * Lines 491-541: p5.prototype.createDiv
   * Lines 543-576: p5.prototype.createP
   * Lines 578-647: p5.prototype.createSpan
   * Lines 649-722: p5.prototype.createImg
   * Lines 724-785: p5.prototype.createA
   * Lines 405-476: input
   * Lines 787-922: p5.prototype.createslider
   * Lines 924-1006: p5.prototype.createButton
   * Lines 1008-1157: p5.prototype.createCheckbox
   * Lines 1159-1438: p5.prototype.createSelect
   * Lines 1440-1739: p5.prototype.createRadio
   * Lines 1741-1839: p5.prototype.createColorPicker
   * Lines 1841-1922: p5.prototype.createInput
   * Lines 1924-2048: p5.prototype.createFileInput
* File 2: p5.MediaElements.js (video/audio/camera functions)
   * Lines 2050-2095: function createMedia
   * Lines 2097-2191: p5.prototype.createVideo
   * Lines 2193-2239: p5.prototype.createAudio
   * Lines 2241-2275: cameraSetup
   * Lines 2277-2472: p5.prototype.createCapture
   * Lines 2475-2531: p5.prototype.createElement (Can also be placed with _validateParameters)
   * Lines 4077-5433: p5.MediaElementAdditions
* File Three: p5.Element.js
   * Lines 330-403: p5.element.prototype.changed
   * Lines 2533-4075: p5.element.prototype. functions
* File Four: p5.File.js
   * Lines 5436-5804 Includes:
   * File class

Please let me know what you all think. @Qianqianye and @limzykenneth please let me know what you think and if you have any suggestion. If I misunderstood either of you and went in the wrong direction let me know and I'll make the necessary corrections.

### Pros (updated based on community comments)

* **Readability:** This proposal makes the source code more readable and manageable by grouping and separating functions in to smaller, relevant files.

### Cons (updated based on community comments)

None so far.

### Proposal status

Under review

### PR DESCRIPTION
**Pull request by SilasVM**
Resolves #7308 

 Changes:

Distributed the code within [dom.js](https://github.com/processing/p5.js/blob/main/src/dom/dom.js), across dom.js, p5.Element.js, p5.MediaElements.js, and p5.File.js.
Specific details of the split are detailed in Proposal #7308.

I've had trouble running the `npm test` command. I was unable to run it on a clean branch, so I don't think this is a direct indication of any issues with my changes. **Please let me know if there's an alternate form of testing for this branch**

I was also unsure of how to modify lines 13-19 of each file. 
- Should dom.js import each new file? 
- Do lines 13-16 need modification? 

 Screenshots of the change:

image

#### PR Checklist

- [x] `npm run lint` passes
- [x] [Inline documentation] is included / updated
- [x] [Unit tests] are included / updated

[Inline documentation]: https://github.com/processing/p5.js/blob/main/contributor_docs/inline_documentation.md
[Unit tests]: https://github.com/processing/p5.js/tree/main/contributor_docs#unit-tests

Please let me know your thoughts @limzykenneth.

### CODE REVIEW

**Comment by davepagurek** [comment]
Hey @SilasVM, what errors did you get when running `npm run test`?

**Comment by davepagurek** [comment]
> Do lines 13-16 need modification?

What you've got now looks good to me!

> Should dom.js import each new file?

Currently the dom module is imported here, which we'll need to modify:

https://github.com/processing/p5.js/blob/39aab8300a71b66f490c6be067eb99b30e9abf6b/src/app.js#L42-L43

Other modules that we've refactored now export a function that gets called to set it up, e.g.:
https://github.com/processing/p5.js/blob/39aab8300a71b66f490c6be067eb99b30e9abf6b/src/app.js#L39-L40

So we'll probably want to do that here too, following a similar pattern to other modules, which [register an addon](https://github.com/processing/p5.js/blob/dev-2.0/src/data/index.js) in their exported setup function, and then [structure the implementation files like addons](https://github.com/processing/p5.js/blob/dev-2.0/src/data/local_storage.js) where we add properties to the `fn` parameter as a shortcut for adding to `p5.prototype`.

**Comment by SilasVM** [comment]
> Hey @SilasVM, what errors did you get when running `npm run test`?

Thank you for the suggestion @davepagurek!

Currently I'm getting a lot of errors surrounding the DOM module. Some are saying "should be function" and it seems some of the  methods aren't returning output as they should.
image

I modified [app.js](https://github.com/processing/p5.js/blob/39aab8300a71b66f490c6be067eb99b30e9abf6b/src/app.js) which reduced the amount of errors I was getting.] from 7 files and 93 failed tests to 5 files and 48 tests failed.
image

When running `npm test` on a clean fork of dev-2.0 I'm seeing 4 files and 10 test failed(I'd appreciate it if someone could test this as well to see if these failed tests are a result of a lacking dependency or some error occurring on my device): 

clean_dev-2 0_npm_test

This is my output when running it on my pr fork after updating app.js, and the first screenshot in this comment is a more detailed look into what's failing:

pr_dev-2 0_npm_test_updated_app

**Comment by limzykenneth** [comment]
@SilasVM For the linter error, please run `npm run docs` first and that should clear it. I'll check the tests.

**Comment by SilasVM** [comment]
Thank you @limzykenneth, the error was cleared.

`npm test` now displays all clear tests within the DOM module:
image

These are the remaining error spots from `npm test`:
image

More detailed looks at the first 3 files:

image
The 4th file:

image

File: src/app.js [hunk 1]
```diff
@@ -40,7 +40,8 @@ import data from './data';
 data(p5);
 
 // DOM
-import './dom/dom';
+import DOM from './dom';
+DOM(p5);
```
Comment (limzykenneth) [inline src/app.js:44]:
```suggestion
import dom from './dom';
dom(p5);
```

File: src/app.js [hunk 2]
```diff
@@ -76,5 +77,6 @@ import './webgl/p5.RendererGL.Retained';
 import './webgl/p5.Texture';
 
 import './core/init';
+import dom from './dom/dom';
```
Comment (limzykenneth) [inline src/app.js:80]: Don't think this line is necessary?
```suggestion
```

File: src/dom/index.js [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+import element from './element';
+import media from './media_element';
+import file from './p5.File';
+
+export default function(p5){
+    p5.registerAddon(dom);
+    p5.registerAddon(element);
+    p5.registerAddon(media);
+    p5.registerAddon(file);
+  }
```
Comment (limzykenneth) [inline src/dom/index.js:11]:
```suggestion
  p5.registerAddon(dom);
  p5.registerAddon(element);
  p5.registerAddon(media);
  p5.registerAddon(file);
}
```

**Review by limzykenneth — CHANGES_REQUESTED** [review]

File: src/app.js (hunk 2 above)
Comment (SilasVM) [reply src/app.js:80]: I tested this removal locally and it's not necessary.

**Comment by SilasVM** [comment]
@limzykenneth, when I was implementing your requests, I noticed the following files have this unnecessary line of code: `import p5 from '../core/main';`.
image

I removed the line in each of the 4 files, ran `npm test`, and everything in DOM passes as it should with no new test failures. I have the branch ready to push, would you advise leaving this line in any of the files for any reason or should I go ahead and update the PR with this modification?

**Comment by limzykenneth** [comment]
Yes these are not typically necessary with the new module syntax.

**Comment by SilasVM** [comment]
Thank you @limzykenneth and @Qianqianye for this opportunity and your assistance throughout!

**Pull request merged**